pontoon/base/forms.py

```diff
@@ -397,3 +398,32 @@ class AddCommentForm(forms.Form):
     entity = forms.IntegerField(required=False)
     comment = HtmlField()
     translation = forms.IntegerField(required=False)
+
+
+class CreateTokenForm(forms.ModelForm):
+    """
+    Form for creating Personal Access Tokens.
+    """
+
+    name = forms.RegexField(
+        regex=r"^[a-zA-Z0-9-]+$",
+        error_messages={
+            "invalid": "Name may only contain letters, numbers, and hyphens."
+        },
+    )
+
+    class Meta:
+        model = PersonalAccessToken
+        fields = ("name",)
+
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop("user", None)
+        super().__init__(*args, **kwargs)
+
+    def clean_name(self):
+        if not self.user:
+            raise ValidationError("User must be provided to create a token.")
```

Comment:
Not sure there's a need to validate that on the form level - data model already takes care of it.